### PR TITLE
mclag enhancements as per HLD at Azure/SONIC#596

### PR DESCRIPTION
**Pull request by gitsabari**
What I did
This PR contains the code changes to support management framework changes for mclag enhancements as per HLD Azure/SONiC#596

Why I did it
Implemented as per the MCLAG enhancements HLD: Azure/SONiC#596

How I verified it
Mclag specific VS tests included part of check-in.

Details:
The PR includes management framework changes part of MCLAG enhancements
This PR must work with submitted PR's in other sonic repositories which are listed below.
        > Iccpd: https://github.com/Azure/sonic-buildimage/pull/4819 
	> Swss: https://github.com/Azure/sonic-swss/pull/1331                                                                    
	> swss-common: https://github.com/Azure/sonic-swss-common/pull/405 
	> Mclagsyncd: https://github.com/Azure/sonic-swss/pull/1349                                           
	> Utilities: https://github.com/Azure/sonic-utilities/pull/1138 
	> Mgmt-FW: https://github.com/Azure/sonic-mgmt-framework/pull/59

### CODE REVIEW

**Comment by prvattem** [comment]
I am fine with changes in this PR. 
Are there any Test Results to be attached to this PR?

File: translib/transformer/xfmr_mclag.go [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+	log "github.com/golang/glog"
+	"github.com/openconfig/ygot/ygot"
+	"strconv"
+	"strings"
+	"translib/db"
+	"translib/ocbinds"
+)
+
+func init() {
+	XlateFuncBind("YangToDb_mclag_domainid_fld_xfmr", YangToDb_mclag_domainid_fld_xfmr)
```
Comment (rathnasabapathyv) [inline translib/transformer/xfmr_mclag.go:31]: Why these field-transformers like (YangToDb_mclag_domainid_fld_xfmr) is required, since openconfig-mclag-annot.yang is not part of this PR? Also couldn't see any other open PR for that OC-yang mclag extension. Also KLISH cmds in other PR (https://github.com/Azure/sonic-mgmt-framework/pull/59) uses sonic-yang for configuration

File: translib/transformer/xfmr_mclag.go [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+var DbToYang_mclag_domain_oper_status_fld_xfmr FieldXfmrDbtoYang = func(inParams XfmrParams) (map[string]interface{}, error) {
+	var err error
+	result := make(map[string]interface{})
+	log.Infof("DbToYang_mclag_domain_oper_status_fld_xfmr --> key: %v", inParams.key)
+
+	stDb := inParams.dbs[db.StateDB]
+	mclagEntry, _ := stDb.GetEntry(&db.TableSpec{Name: "MCLAG_TABLE"}, db.Key{Comp: []string{inParams.key}})
+	operStatus := mclagEntry.Get("oper_status")
+	if operStatus == "up" {
+		result["oper-status"], _ = ygot.EnumName(ocbinds.OpenconfigMclag_Mclag_MclagDomains_MclagDomain_State_OperStatus_OPER_UP)
```
Comment (rathnasabapathyv) [inline translib/transformer/xfmr_mclag.go:86]: Enums like these "OpenconfigMclag_Mclag_MclagDomains_MclagDomain_State_OperStatus_OPER_UP" may not compile without openconfig-mclag.yang, since ocbinds.go will not generate enums for these. Also in all other similar places like these, we will get compilation error. If openconfig-mclag.yang is going to come as part of other PR, then these code can come later as well. Thoughts?

File: translib/transformer/xfmr_mclag.go [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+
+package transformer
+
+import (
+	log "github.com/golang/glog"
+	"github.com/openconfig/ygot/ygot"
+	"strconv"
+	"strings"
+	"translib/db"
+	"translib/ocbinds"
```
Comment (prvattem) [inline translib/transformer/xfmr_mclag.go:27]: Are you using openconfig-yang models?

File: translib/transformer/xfmr_mclag.go (hunk 2 above)
Comment (gitsabari) [reply translib/transformer/xfmr_mclag.go:86]: will update the oc yang part of this PR. Thanks

File: translib/transformer/xfmr_mclag.go (hunk 3 above)
Comment (gitsabari) [reply translib/transformer/xfmr_mclag.go:27]: yes, will udpate oc yang along with this PR

File: translib/transformer/xfmr_mclag.go (hunk 2 above)
Comment (rathnasabapathyv) [reply translib/transformer/xfmr_mclag.go:86]: OK then will wait for those changes, before approving this PR

Comment (gitsabari) [reply translib/transformer/xfmr_mclag.go:86]: rathnasabapathyv, i have updated oc yang along with this PR please go ahead and look at it and approve it. thanks

**Comment by gitsabari** [comment]
since sonic-yang files got moved to sonic-buildimage removing this PR as this doesn't hold good
